api/analyzers/python/analyzer.py

```
@@ -0,0 +1,124 @@
+
+import logging
+logger = logging.getLogger('code_graph')
+
+class PythonAnalyzer(AbstractAnalyzer):
+    def __init__(self) -> None:
+        super().__init__(Language(tspython.language()))
+    
+    def add_dependencies(self, path: Path, files: list[Path]):
+        if Path(f"{path}/venv").is_dir():
+            return
+        subprocess.run(["python3", "-m", "venv", "venv"], cwd=str(path))
```

Ruff (0.15.5)

[error] Starting a process with a partial executable path

(S607)